packages/react/src/components/Storage/StorageManager/DropZone/DropZone.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import { View, ComponentClassNames, Text } from '../../../../primitives';
+import { classNameModifier } from '../../../../primitives/shared/utils';
+import { IconUpload } from '../../../../primitives/Icon/internal';
+import { DropZoneProps } from './types';
+import { FilePicker } from './FilePicker';
+
+export function DropZone({
+  onChange,
+  isLoading = false,
+  dropFilesText,
+  browseFilesText,
+  acceptedFileTypes,
+}: DropZoneProps): JSX.Element {
+  const [inDropZone, setInDropZone] = useState(false);
+
+  const onDragStart = (event: React.DragEvent<HTMLDivElement>) => {
```

Comment:
Move dropzone handlers to a hook called from StorageManager to make it easier to override the StorageManager